src/gui/plugins/scene3d/Scene3D.cc

```diff
@@ -215,6 +225,9 @@ inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
     /// \brief Flag for indicating whether we are in view angle mode or not
     public: bool viewAngle = false;
 
+    /// \brief Flag for indicating whether we are in move to pose mode or not
+    public: bool moveToPose = false;
```

Comment:
Maybe we don't need this flag if we set the `moveToPoseValue` to NaNs when not in move to pose mode. Just a suggestion.

Reply:
Removed the flag, and used an `std::optional` in 251b573c3e83ee444d4d136a08eeb5e514eae91f